### ISSUE TITLE
Seemingly-incorrect error for `if let ... && ...`

### ISSUE THREAD

**Issue by camelid** [rust-lang/rust#78205]
```rust
fn main() {
    if let x = 3 && true {}
}
```

([Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=30d5444bc12d788672013143e96a65ca))

The second note says "as well as when nested within `&&` [...]" – isn't that what I did in the code?

```
   Compiling playground v0.0.1 (/playground)
error: `let` expressions are not supported here
 --> src/main.rs:4:8
  |
4 |     if let x = 3 && true {}
  |        ^^^^^^^^^
  |
  = note: only supported directly in conditions of `if`- and `while`-expressions
  = note: as well as when nested within `&&` and parenthesis in those conditions

error: aborting due to previous error

error: could not compile `playground`

To learn more, run the command again with --verbose.
```

**Comment by camelid** [comment]
Note that the error occurs without the feature flag on nightly as well (but not on stable):

```
error: `let` expressions are not supported here
 --> src/main.rs:2:8
  |
2 |     if let x = 3 && true {}
  |        ^^^^^^^^^
  |
  = note: only supported directly in conditions of `if`- and `while`-expressions
  = note: as well as when nested within `&&` and parenthesis in those conditions
```

**Comment by c410-f3r** [comment]
This issue can be closed

**Comment by camelid** [comment]
Yeah, looks like the error is better now:

```
error[E0658]: `let` expressions in this position are unstable
 --> src/main.rs:2:8
  |
2 |     if let x = 3 && true {}
  |        ^^^^^^^^^
  |
  = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
  = help: add `#![feature(let_chains)]` to the crate attributes to enable

For more information about this error, try `rustc --explain E0658`.
error: could not compile `playground` due to previous error
```

And with the feature-gate, it just works.
